### PR TITLE
Forgot password validation

### PR DESCRIPTION
**Pull request by JessicaJohnBritto**
Have included forgot password feature and password validation in the register form and included conditions for several cases in login and register forms.

### CODE REVIEW

File: register/forms.py [hunk 1]
```diff
@@ -1,16 +1,96 @@
+# # from django import forms
```
Comment (ExpressHermes) [inline register/forms.py]: remove the comments

File: register/models.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
 from django.db import models
-
+from django.contrib.auth.models import AbstractUser
```
Comment (ExpressHermes) [inline register/models.py]: remove the import if you are not using it

File: register/templates/register/base.html [hunk 3]
```diff
@@ -7,6 +7,17 @@
     <title> Helping Hand | Register | Login </title>
     <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous">
     <link rel="stylesheet" type="text/css" href="{% static 'css/style.css' %}">
+    <style>
```
Comment (ExpressHermes) [inline register/templates/register/base.html]: put the styles in the `styles.css` file

File: register/templates/register/forgot_password.html [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+
+
+
+
+
+
+
+
+{%comment%}
```
Comment (ExpressHermes) [inline register/templates/register/forgot_password.html]: this is commented code, right? please all the commented code if it's not relevant

File: register/templates/register/register.html [hunk 5]
```diff
@@ -11,22 +12,61 @@
     </button>
   </div>
   {% endif %}
-  {% csrf_token %}
+  <!-- {% for error in errors%}
```
Comment (ExpressHermes) [inline register/templates/register/register.html]: remove comments

File: register/views.py [hunk 6]
```diff
@@ -13,29 +15,43 @@
...
+        
         if user_form.is_valid():
             user = user_form.save()
-            user.set_password(user.password)
+            password1 = user_form.cleaned_data.get('password1')
+            # password2 = user_form.cleaned_data.get('password2')
+            # print(password1)
+            user.last_name=request.POST['lname']
+            user.first_name=request.POST['fname']
+            print("##########################################################################")
```
Comment (ExpressHermes) [inline register/views.py]: remove the print statements

File: register/views.py [hunk 7]
```diff
@@ -13,29 +15,43 @@
...
             user.save()
-
             # encoding user id to send over the network
 
             uidb64 = urlsafe_base64_encode(force_bytes(user.pk))
-            domain = get_current_site(request).domain
+            # Use the code below to get domain name
+            # domain = get_current_site(request).domain
+            # use the code below while running on local server since the above code is not changing is_active to True
+            domain='127.0.0.1:8000'
```
Comment (ExpressHermes) [inline register/views.py]: this is the hardcoded domain. Use `get_current_site(request).domain` to generate domain dynamically

File: register/views.py [hunk 8]
```diff
@@ -85,24 +101,153 @@ def user_verify(request, uidb64, token):
...
+                return render(request, 'register/login.html', {'error': 'You have not activated your account, activate using the link sent to your mail.'})
         else:
-            messages.warning(request, 'You need to register yourself first!')
-            return redirect('/register/')
+            try:
+                user=User.objects.get(username=username)
+                if not user.is_active:
+                    messages.warning(request, 'You have not activated your account, activate using the link sent to your mail.')
+                    return render(request, 'register/login.html', {'error': 'You have not activated your account, activate using the link sent to your mail.'})
+                if password!=user.password:
```
Comment (ExpressHermes) [inline register/views.py]: this is not how the password is compared. Use [this SO answer](https://stackoverflow.com/a/55822545/13398404) and use `check_password()` function to do it properly.

File: register/views.py [hunk 9]
```diff
@@ -85,24 +101,153 @@ def user_verify(request, uidb64, token):
...
+                print("Not a registered user")
+                messages.warning(request, 'You need to register yourself first!')
+                return redirect('/register/')
     else:
-        return render(request, 'register/login.html')
+        return render(request, "register/login.html")
+
+def forgot_password(request):
+    User=get_user_model()
+    print("Not post")
```
Comment (ExpressHermes) [inline register/views.py]: remove the print statements, please

**Review by ExpressHermes — CHANGES_REQUESTED** [review]
Some of the things to note:
- Remove necessary code, comments, and packages.
- Format code (Use prettier for js, and HTML and black/yapf for python file)

File: register/views.py (hunk 7 above)
Comment (JessicaJohnBritto) [reply register/views.py]: Can I comment out the code on line 50? or should I remove it? @ExpressHermes

**Review by ExpressHermes — APPROVED** [review]

**Pull request merged**